Miscellaneous fixups

GiudGiud opened the pull request:
Things I ran into during Easter weekend MaCaw dev

GiudGiud committed:
Add nonlinear system parameter group to FEProblemBase, refs #20535

GiudGiud committed:
Allow another ray tracing failure in devel mode if the study said so
refs #16028

GiudGiud committed:
Adjustments to verbosity of mesh generators

GiudGiud committed:
Adapt parameter groups for mesh generator params to avoid users trying to use an advanced feature
Improve save_in_mesh docstring

GiudGiud committed:
Fix up mesh saving capability tests, in relation #23618

GiudGiud committed:
Expand GiudGiud's area of awareness, refs #0

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1453023/) on da8b1cb wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/24023/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1453037/) on da8b1cb wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>aa09c9</th>
      <th colspan=4>#24023 da8b1c</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.01%</td>
      <td>85.01%</td>
      <td>+0.00%</td>
      <td>47.06%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>85482</td>
      <td>85483</td>
      <td>+1</td>
      <td>8</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15077</td>
      <td>15077</td>
      <td>-</td>
      <td>9</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/24023/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/24023/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/24023/coverage/xfem)

</details>

## Warnings

- `framework` new line coverage rate 47.06% is less than the suggested 90.0%



This comment will be updated on new commits.

lindsayad commented on framework/src/meshgenerators/PatternedMeshGenerator.C:
```diff
@@ -310,8 +310,8 @@ PatternedMeshGenerator::mergeSubdomainNameMaps(
       paramError("inputs",
-                 "The input meshes both contain subdomain name " + name_to_insert +
-                     " that correspond to conflicting subdomain ids.");
+                 "Two of the input meshes contain a subdomain with then name '" + name_to_insert +
```
```suggestion
                 "Two of the input meshes contain a subdomain with the name '" + name_to_insert +
```

lindsayad commented on framework/src/problems/FEProblemBase.C:
```diff
@@ -273,8 +273,11 @@ FEProblemBase::validParams()
+      "material_dependency_check check_uo_aux_state error_on_jacobian_nonzero_reallocation",
       "Simulation checks");
+  params.addParamNamesToGroup("use_nonlinear previous_nl_solution_required nl_sys_names "
+                              "error_on_jacobian_nonzero_reallocation ignore_zeros_in_jacobian",
```
Is there any problem with having the same parameter in two different groups?

lindsayad replied:
(Referring to `error_on_jacobian_nonzero_reallocation`)

GiudGiud replied:
seems it just gets ignored the second time. I ll keep it only once

lindsayad commented on test/tests/mesh/save_in_meshes/tests:
If modifying these details, can you add the commas and such to make the whole requirement plus details a "well-structured" sentence?

GiudGiud replied:
done

GiudGiud committed:
Address review:
- better requirement sentence structure
Update framework/src/meshgenerators/PatternedMeshGenerator.C

Co-authored-by: Alex Lindsay <alexlindsay239@gmail.com>
